public/app/features/manage-dashboards/state/actions.ts

```diff
@@ -0,0 +1,164 @@
+import { DataSourceInstanceSettings } from '@grafana/data';
+import { getBackendSrv } from '@grafana/runtime';
+import config from 'app/core/config';
+import {
+  dashboardTitleChange,
+  clearDashboard,
+  setInputs,
+  setGcomDashboard,
+  setJsonDashboard,
+  setGcomError,
+  dashboardUidChange,
+  dashboardUidExists,
+  dashboardTitleExists,
+} from './reducers';
+import validationSrv from '../services/ValidationSrv';
+import { ThunkResult } from 'app/types';
+import { updateLocation } from '../../../core/actions';
+import locationUtil from '../../../core/utils/location_util';
+
+export function fetchGcomDashboard(id: string): ThunkResult<void> {
+  return async dispatch => {
+    try {
+      const dashboard = await getBackendSrv().get(`/api/gnet/dashboards/${id}`);
+      dispatch(setGcomDashboard(dashboard));
+      dispatch(processInputs(dashboard.json));
+    } catch (error) {
+      dispatch(setGcomError(error.data.message || error));
+    }
+  };
+}
+
+export function importDashboardJson(dashboard: any): ThunkResult<void> {
+  return async dispatch => {
+    await getBackendSrv()
+      .get(`/api/dashboards/uid/${dashboard.uid}`)
+      .then(result => {
+        dispatch(dashboardUidExists(result));
+      })
+      .catch(error => {
+        error.isHandled = true;
+      });
```

Comment:
An alternative could be to change this to `await` to prevent mixing async/await and promises syntax.

``` typescript
try {
   const result = await getBackendSrv().get(`/api/dashboards/uid/${dashboard.uid}`);
   dispatch(dashboardUidExists(result));
} catch (error) {
    error.isHandled = true; // not really sure why we do this since the error never is used.
}
```

Reply:
the error.isHandled must be there otherwise backend srv will throw a appnotification

Reply:
aha, now I remember. Thanks! :)